### ISSUE TITLE
PUT /certificates: snis is an unknown field

### ISSUE THREAD

**Issue by luispabon** [Kong/kong#2792]
### Summary

I'm writing some automation around deploying SSL certs for specific SNIs on Kong 0.10.3. I'm following the docs here:

https://getkong.org/docs/0.10.x/admin-api/#add-certificate

specifically, 

```
snis  | One or more hostnames to associate with this certificate as an SNI. This is a sugar parameter that will, under the hood, create an SNI object and associate it with this certificate for your convenience.
```

When I send in a certs payload in json with SNIs, I get a `400 / snis is an unknown field` error, like so:

```
curl -XPUT http://kong-admin:8001/certificates -H"accept: application/json" -H"content-type: application/json" -d"
{
    "cert": "-----BEGIN CERTIFICATE-----blah blah-----END CERTIFICATE-----\n",
    "key": "-----BEGIN PRIVATE KEY-----blah blah----END PRIVATE KEY-----\n",
    "snis": [
        "my-sni",
    ]
}"
```

I get the following 400 response:

```json
{
    "snis": "snis is an unknown field"
}
```

The request goes through without the "snis" field. If I request back the cert I just created at the API, I can see the "snis" coming back indeed (blank, obviously).

I realise PUT is not documented above, but it clearly exists and has it's own separate payload behaviout.  I would expect it to be an UPSERT operation.

### Additional Details & Logs

- Kong version: 0.10.3
- Operating System: kubernetes 1.5

**Comment by undernewmanagement** [comment]
I also have the exact problem here. This is my command:

```
curl -X PUT http://localhost:8001/certificates/ \
              -d "cert=@/etc/letsencrypt/live/api.domain.com/fullchain.pem" \
              -d "key=@/etc/letsencrypt/live/api.domain.com/privkey.pem" \
              -d "snis=api.domain.com,portal.domain.com,www.domain.com"
```

My current work-around has been to delete the certs and re-create them.

```
curl -X DELETE http://localhost:8001/certificates/domain.com
curl -X DELETE http://localhost:8001/certificates/portal.domain.com
curl -X DELETE http://localhost:8001/certificates/api.domain.com

curl -X POST http://localhost:8001/certificates/ \
              -d "cert=@/etc/letsencrypt/live/api.domain.com/fullchain.pem" \
              -d "key=@/etc/letsencrypt/live/api.domain.com/privkey.pem" \
              -d "snis=api.domain.com,portal.domain.com,www.domain.com"
```

**Comment by thibaultcha** [comment]
Hi,

This might be an overlook on our side. Under the hood, a `POST /certificates` will actually deal with two underlying entities in Kong: ssl_certificates and snis. 

A standard `PUT` request (when `PUT` is not overridden for a given endpoint), will simply deal with the underlying entity alone, in this case, ssl_certificates. And as a matter of fact, the ssl_certificate entity does *not* have an `snis` field. See its definition [here](https://github.com/Kong/kong/blob/master/kong/dao/schemas/ssl_certificates.lua).

A PR for this would be most welcome, otherwise, we'll add it to our internal backlog for tracking and will get to it when we'll have the time...

Thank you for the report!

**Comment by thibaultcha** [comment]
@hbagdi has contributed #3040 which should take care of this, and it will be included in our upcoming 0.12.0 release!

**Comment by luispabon** [comment]
:+1: :+1: :+1:
